internal/message.go

```diff
@@ -101,3 +104,21 @@ func getRetryCount(message *segmentio.Message) int {
 
 	return 0
 }
+
+func getMessageUnixNanoTime(message *segmentio.Message) int64 {
+	for i := range message.Headers {
+		if message.Headers[i].Key != MessageTimeHeaderKey {
+			continue
+		}
+
+		ts, _ := strconv.Atoi(string(message.Headers[i].Value))
+		return int64(ts)
+	}
+
+	message.Headers = append(message.Headers, segmentio.Header{
```

Comment:
When message time header is not exist, I put with empty value (0) so in the produce side no need to check message time header also exist or not